packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -226,18 +232,42 @@ void updateLocalProperties({String projectPath, BuildInfo buildInfo}) {
     settings.writeContents(localProperties);
 }
 
-Future<Null> findAndReplaceVersionProperties({@required BuildInfo buildInfo}) {
-  assert(buildInfo != null, 'buildInfo can\'t be null');
+Future<Null> findAndReplaceVersion() async {
+  return findAndReplaceVersionInBuildGradle(projectPath: '');
+}
+
+@visibleForTesting
+Future<Null> findAndReplaceVersionInBuildGradle({String projectPath}) async {
+  final Status status = logger.startProgress('Set version in app/build.gradle...', expectSlowOperation: true);
   final Completer<Null> completer = new Completer<Null>();
```

Comment:
No need to use a `Completer` any more -- you're async now.

In review comments below, I suggest a new use for a completer, but even then, I think you should await the completer's future in the body of this method, but not return the completer's future.